README.md

```diff
@@ -47,11 +48,59 @@ public function middleware()
 }
 ```
 
-Finally, you can add your own collector to the agent!
+### Add a collector for other events
+
+You can add extra collector(s) to listen to your own application events or Laravel events like `Illuminate\Mail\Events\MessageSending` for example. We created a base collector that already includes functionality to measure events, that you can extend from:
+
+```php
+// app/Collectors/MailMessageCollector.php
+
+namespace YourApp\Collectors;
+
+use AG\ElasticApmLaravel\Contracts\DataCollector;
+use AG\ElasticApmLaravel\Collectors\EventDataCollector;
+
+use Illuminate\Mail\Events\MessageSending;
+use Illuminate\Mail\Events\MessageSent;
+
+class MailMessageCollector extends EventDataCollector implements DataCollector
+{
+    public function getName(): string
+    {
+        return 'mail-message-collector';
+    }
+
+    protected function registerEventListeners(): void
+    {
+        $this->app->events->listen(MessageSending::class, function (\Swift_Message $message) {
+            $this->startMeasure(
+                'mail #' . $message->getId(),
+                'mail.delivery',
+            );
+        });
+
+        $this->app->events->listen(StopMeasuring::class, function (\Swift_Message $message) {
+            $this->stopMeasure('mail #' . $message->getId());
+        });
+    }
+}
 
 ```
-$agent = app(\AG\ElasticApmLaravel\Agent::class);
-$agent->addCollector(new MyCollector());
+
+Don't forget to register your collector when the application starts:
+
+```php
+// app/Providers/AppServiceProvider.php
+
+use AG\ElasticApmLaravel\Facades\ApmCollector;
+
+use YourApp\Collectors\MailMessageCollector;
+
+public function boot()
```

Comment:
I like this recommendation in boot instead of register and think we should put the default collectors under boot as well.

Reply:
If I understand correctly `register/boot` methods, we should register the agent and facade in the `register` method, and the middleware and collectors in the `boot` method. Am I correct?

Reply:
Yep anything that describes how an object should be instantiated goes in register, and the usages of those objects can be in boot.